test/e2e/infra/secondary_network/secondary_network_prerequisite.sh

```diff
@@ -0,0 +1,360 @@
+        --host-name            	 K8s cluster control node's host name or IP address."
+
+_ssh_config_info="Prerequisite: Kubernetes(K8s) cluster is up and running at the local or remote server.
+
+Follow these steps to configure SSH information to access the kubernetes cluster control node.
+
+1) Update remote/local cluster control node information at your /home/<user>/.ssh/config
+Example:
+Host <Control-Plane-Node>
+    HostName <Control-Plane-IP>
+    Port 22
+    user ubuntu
+    IdentityFile /home/ubuntu/.ssh/id_rsa
+
+2) Add remote/local cluster control node's public key (id_rsa.pub) to /home/<user>/.ssh/authorized_keys
+
+3) Set PubkeyAuthentication at sshd_config to yes at your remote/local cluster control node
+
+4) Esure SSH to the remote/local node works after the above configuration (ssh <control node/host name>)"
+
+function echoerr {
+        echo >&2 "$@"
+}
+
+function print_usage {
+	echoerr "$_usage"
+}
+
+function print_ssh_config_info {
+	echoerr "$_ssh_config_info"
+}
+
+function print_help {
+	echoerr "Try '$0 --help' for more information."
+}
+
+export URL_ANTREA_CONFIG_MAP='https://raw.githubusercontent.com/k8snetworkplumbingwg/sriov-network-device-plugin/master/deployments/configMap.yaml'
+export URL_SRIOV_DP_DAEMONSET='https://raw.githubusercontent.com/k8snetworkplumbingwg/sriov-network-device-plugin/master/deployments/k8s-v1.16/sriovdp-daemonset.yaml'
+export URL_WHEREABOUTS_IP_POOLS='https://raw.githubusercontent.com/k8snetworkplumbingwg/whereabouts/master/doc/crds/whereabouts.cni.cncf.io_ippools.yaml'
+export URL_WHEREABOUTS_OVERLAPPING_IP_RANGE_RES='https://raw.githubusercontent.com/k8snetworkplumbingwg/whereabouts/master/doc/crds/whereabouts.cni.cncf.io_overlappingrangeipreservations.yaml'
+export YAML_CONFIG_MAP='configMap.yaml'
+export YAML_SRIOV_DP_DAEMONSET='sriovdp-daemonset.yaml'
+export YAML_ANTREA='antrea.yml'
+export YAML_WHEREABOUTS_IP_POOL='whereabouts.cni.cncf.io_ippools.yaml'
+export YAML_WHEREABOUTS_OVERLAPPING_IP_RANGE_RES='whereabouts.cni.cncf.io_overlappingrangeipreservations.yaml'
+export YAML_NET_ATTACH_DEF_CRD='network-attachment-definition-crd.yaml'
+export YAML_SEC_NET_CONFIG_FILE='secondary_network_configuration.yaml'
+export YAML_VIRTUAL_NET_INSTANCE_01='virtual-network-instance1.yaml'
+export YAML_VIRTUAL_NET_INSTANCE_02='virtual-network-instance2.yaml'
+export YAML_VIRTUAL_NET_INSTANCE_03='virtual-network-instance3.yaml'
+#SEC_NET_LOCAL_CONFIG_DIR_PATH is at the test execution server where the required config files are downloaded/generated.
+export SEC_NET_LOCAL_CONFIG_DIR_PATH=$THIS_DIR/secondary_network_config/
+#SEC_NET_REMOTE_CONFIG_DIR_PATH is at the K8s cluster's control node where we configure the cluster for secondary network.
+export SEC_NET_REMOTE_CONFIG_DIR_PATH='/tmp/secondary_network_config/'
+export NAMESPACE='kube-system'
+
+interface=''
+vfs_num=''
+action=''
+configure=''
+
+function validateOptions() {
```

Comment:
it seems that `--host-name` is required for this script to work, maybe we should therefore check that `hostIP` is not empty?

Reply:
Yes, hostIP is mandatory. we are checking it already as part of the parameter verification (case statement). like, if --hostIP is provided, then we look for its $2.

Reply:
that's not what I meant, but I see that you have the `$host_name_set != true` check below now